ci-operator/step-registry/ipi/conf/azurestack/ipi-conf-azurestack-commands.sh

```diff
@@ -24,10 +24,19 @@ echo $master_type
 
 ENDPOINT="${AZURESTACK_ENDPOINT}"
 echo "ASH ARM Endpoint: ${ENDPOINT}"
+
 cp "/var/run/azurestack-cluster-secrets/service-principal" "${SHARED_DIR}/osServicePrincipal.json"
+if [[ -f "${CLUSTER_PROFILE_DIR}/qe-osServicePrincipal.json" ]]; then
```

Comment:
I guess we can use `osServicePrincipal.json` as the key name in vault.

Reply:
could not, because same key osServicePrincipal.json also exists for clusterprofile `azurestack`, and the content is different than credentials `service-principal` mounted in the step, so I changed key name to `qe-osServicePrincipal.json` in vault.

Reply:
> same key osServicePrincipal.json also exists for clusterprofile `azurestack`

The content of `osServicePrincipal.json` should vary accordingly per the specified cluster file, so when we are using  `azurestack-qe` cluster profile, that is expected the content  of `osServicePrincipal.json`  get changed.

for `service-principal` mounted via secret, I am not sure its backgroup, but seem like we did not change its original logic yet.

Reply:
Okay, if your comments already take https://github.com/openshift/release/pull/34523#discussion_r1047969799 into consideration, yes, it indeed bring some problem.

Reply:
good idea, changed to use cloud_name for condition, and update to osServicePrincipal.json in vault.